From: yang <yang@fydeos.io>
Date: Thu, 20 Jan 2022 13:32:01 +0800
Subject: fix vmware es2.1 to es2.0
Signed-off-by: Fangzhou Chen <fangzhou@fydeos.io>

---

Index: src/gpu/command_buffer/service/feature_info.cc
===================================================================
--- src.orig/gpu/command_buffer/service/feature_info.cc
+++ src/gpu/command_buffer/service/feature_info.cc
@@ -492,11 +492,13 @@ void FeatureInfo::InitializeFeatures() {
 
   // Really it's part of core OpenGL 2.1 and up, but let's assume the
   // extension is still advertised.
+  /*
   bool has_pixel_buffers =
       gl_version_info_->is_es3 || gl_version_info_->is_desktop_core_profile ||
       gfx::HasExtension(extensions, "GL_ARB_pixel_buffer_object") ||
       gfx::HasExtension(extensions, "GL_NV_pixel_buffer_object");
-
+  */
+  bool has_pixel_buffers = false;
   // If ES3 or pixel buffer objects are enabled by the driver, we have to assume
   // the unpack buffer binding may be changed on the underlying context. This is
   // true whether or not this particular decoder exposes PBOs, as it could be
