Index: mesa-freedreno/src/gallium/drivers/svga/svga_format.c
===================================================================
--- mesa-freedreno.orig/src/gallium/drivers/svga/svga_format.c
+++ mesa-freedreno/src/gallium/drivers/svga/svga_format.c
@@ -271,6 +271,8 @@ svga_translate_screen_target_format_vgpu
    switch (format) {
    case PIPE_FORMAT_B8G8R8A8_UNORM:
       return SVGA3D_B8G8R8A8_UNORM;
+   case PIPE_FORMAT_R8G8B8A8_UNORM:
+      return SVGA3D_R8G8B8A8_UNORM;
    case PIPE_FORMAT_B8G8R8X8_UNORM:
       return SVGA3D_B8G8R8X8_UNORM;
    case PIPE_FORMAT_B5G6R5_UNORM:
@@ -2244,6 +2246,7 @@ svga_is_format_supported(struct pipe_scr
       /* VGPU10 formats */
       case SVGA3D_B8G8R8A8_UNORM:
       case SVGA3D_B8G8R8X8_UNORM:
+      case SVGA3D_R8G8B8A8_UNORM:
       case SVGA3D_B5G6R5_UNORM:
       case SVGA3D_B8G8R8X8_UNORM_SRGB:
       case SVGA3D_B8G8R8A8_UNORM_SRGB:
