diff --git a/Android.mk b/Android.mk
index 9169cef..7df8cbd 100644
--- a/Android.mk
+++ b/Android.mk
@@ -26,7 +26,8 @@ MINIGBM_SRC := \
 	udl.c \
 	vc4.c \
 	vgem.c \
-	virtio_gpu.c
+	virtio_gpu.c \
+    vmwgfx.c
 
 MINIGBM_CPPFLAGS := -std=c++14
 MINIGBM_CFLAGS := \
diff --git a/drv.c b/drv.c
index 0976404..a91bd6b 100644
--- a/drv.c
+++ b/drv.c
@@ -61,6 +61,9 @@ extern const struct backend backend_udl;
 #ifdef DRV_VC4
 extern const struct backend backend_vc4;
 #endif
+#ifdef DRV_VMWGFX
+extern const struct backend backend_vmwgfx;
+#endif
 extern const struct backend backend_vgem;
 extern const struct backend backend_virtio_gpu;
 
@@ -111,6 +114,9 @@ static const struct backend *drv_get_backend(int fd)
 #ifdef DRV_VC4
 		&backend_vc4,
 #endif
+#ifdef DRV_VMWGFX
+        &backend_vmwgfx,
+#endif
 		&backend_vgem,     &backend_virtio_gpu,
 	};
 
diff --git a/vmwgfx.c b/vmwgfx.c
new file mode 100644
index 0000000..fa8107a
--- /dev/null
+++ b/vmwgfx.c
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 The FydeOS Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#ifdef DRV_VMWGFX
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <vmwgfx_drm.h>
+#include <xf86drm.h>
+
+#include "drv_priv.h"
+#include "helpers.h"
+#include "util.h"
+
+static const uint32_t render_target_formats[] = { DRM_FORMAT_ARGB8888, DRM_FORMAT_RGB565,
+                          DRM_FORMAT_XRGB8888, DRM_FORMAT_XRGB1555 };
+
+static int vmwgfx_init(struct driver *drv)
+{
+    drv_add_combinations(drv, render_target_formats, ARRAY_SIZE(render_target_formats),
+                 &LINEAR_METADATA, BO_USE_RENDER_MASK);
+
+    return drv_modify_linear_combinations(drv);
+}
+
+const struct backend backend_vmwgfx = {
+    .name = "vmwgfx",
+    .init = vmwgfx_init,
+    .bo_create = drv_dumb_bo_create,
+    .bo_import = drv_prime_bo_import,
+    .bo_destroy = drv_dumb_bo_destroy,
+    .bo_map = drv_dumb_bo_map,
+    .bo_unmap = drv_bo_munmap,
+};
+
+#endif
